src/java/org/apache/cassandra/net/OutboundTcpConnection.java

```diff
@@ -505,6 +508,10 @@ private boolean connect()
             {
                 logger.error("SSL handshake error for outbound connection to " + socket, e);
                 socket = null;
+                if (DatabaseDescriptor.isSslNteEnabled()) {
```

Comment:
I wonder if this will throw even for Connection timeouts/etc? Probably should test a rolling restart with this.

Reply:
We should try to force timeout by issuing a range scan or making the query timeout window unreasonably small to make sure we treat timeouts as intended.

Reply:
Yup confirmed on IL that this is ok, and I investigated what could cause SSLHandshakeErrors, and this should be safe.